### ISSUE TITLE
ClassNotFoundError: scala/Product$class

### ISSUE THREAD

**Issue by fommil** [lucidsoftware/neo-sbt-scalafmt#16]
on this exact version of my project https://github.com/fommil/drone-dynamic-agents/tree/2152a66f60d1b78df043fb9a4c7a209d077941eb

converting sbt-scalariform to sbt-scalafmt and issuing `sbt compile:scalafmt` gives this `NoClassDefFoundError`.

```
> scalafmt
[info] Formatting 9 Scala sources in drone-dynamic-agents:compile ...
[trace] Stack trace suppressed: run 'last compile:scalafmt' for the full output.
[error] (compile:scalafmt::scalafmt) java.lang.NoClassDefFoundError: scala/Product$class
[error] Total time: 1 s, completed 18-Jun-2017 11:13:00
> last scalafmt
[debug] /home/fommil/Projects/drone-dynamic-agents/.scalafmt.conf does not exist
[info] Formatting 9 Scala sources in drone-dynamic-agents:compile ...
java.lang.NoClassDefFoundError: scala/Product$class
	at org.scalafmt.Versions$.<init>(Versions.scala:6)
	at org.scalafmt.Versions$.<clinit>(Versions.scala)
	at org.scalafmt.config.ScalafmtConfig$.apply$default$1(ScalafmtConfig.scala:2)
	at org.scalafmt.config.Settings$class.$init$(Settings.scala:20)
	at org.scalafmt.config.ScalafmtConfig$.<init>(ScalafmtConfig.scala:49)
	at org.scalafmt.config.ScalafmtConfig$.<clinit>(ScalafmtConfig.scala)
	at org.scalafmt.config.Config$.fromHocon(Config.scala:42)
	at com.lucidchart.scalafmt.impl.ScalafmtFactory.fromConfig(ScalafmtFactory.scala:12)
	at com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin$autoImport$$anonfun$6$$anonfun$apply$3.apply(ScalafmtCorePlugin.scala:56)
	at com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin$autoImport$$anonfun$6$$anonfun$apply$3.apply(ScalafmtCorePlugin.scala:54)
	at com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin$autoImport$$anonfun$11.scalafmtter$lzycompute$1(ScalafmtCorePlugin.scala:91)
	at com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin$autoImport$$anonfun$11.com$lucidchart$sbt$scalafmt$ScalafmtCorePlugin$autoImport$$anonfun$$scalafmtter$2(ScalafmtCorePlugin.scala:91)
	at com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin$autoImport$$anonfun$11$$anonfun$17$$anonfun$apply$11.apply(ScalafmtCorePlugin.scala:120)
	at com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin$autoImport$$anonfun$11$$anonfun$17$$anonfun$apply$11.apply(ScalafmtCorePlugin.scala:118)
	at scala.util.Either$LeftProjection.flatMap(Either.scala:364)
	at com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin$autoImport$$anonfun$11$$anonfun$17.apply(ScalafmtCorePlugin.scala:118)
	at com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin$autoImport$$anonfun$11$$anonfun$17.apply(ScalafmtCorePlugin.scala:118)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:47)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractTraversable.map(Traversable.scala:105)
	at com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin$autoImport$$anonfun$11.apply(ScalafmtCorePlugin.scala:118)
	at com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin$autoImport$$anonfun$11.apply(ScalafmtCorePlugin.scala:82)
```

**Comment by pauldraper** [comment]
It's a bug (or at least a difference) in coursier: https://github.com/coursier/coursier/issues/543

You need to replace `sbt-scalafmt` with `sbt-scalafmt-coursier`. https://github.com/lucidsoftware/neo-sbt-scalafmt#coursier-beta

Coursier seems to be very popular; this bug has been reported a few times. There should probably be runtime check to detect this problem and recommend the solution. And even clearer documentation.

**Comment by pauldraper** [comment]
Addressed in #21

**Comment by fommil** [comment]
bizarrely, I still get this on a work project... but can't reproduce at home.
